provision/pool/pool_test.go

```diff
@@ -39,17 +42,6 @@ func (s *S) TearDownSuite(c *check.C) {
 	s.storage.Close()
 }
 
-func (s *S) SetUpTest(c *check.C) {
```

Comment:
SetUpTest should continue to exist with a call to `dbtest.ClearAllCollections(s.storage.Apps().Database)` to ensure a clean db on each test.
Also, the database:name in `SetUpSuite` should be changed not to conflict with the database:name used in `provision/suite_test.go`, because tests from different packages run in parallel.